src/expr/symbol_manager.cpp

```diff
@@ -57,12 +65,19 @@ class SymbolManager::Implementation
   TermStringMap d_names;
   /** The set of terms with assertion names */
   TermSet d_namedAsserts;
+  /**
+   * Have we pushed a scope (e.g. a let or quantifier) in the current context?
+   */
+  CDO<bool> d_hasPushedScope;
 };
 
 bool SymbolManager::Implementation::setExpressionName(api::Term t,
                                                       const std::string& name,
                                                       bool isAssertion)
 {
+  // cannot name subexpressions under quantifiers
+  PrettyCheckArgument(
+      !d_hasPushedScope.get(), name, "cannot name function in a scope");
```

Comment:
Isn't this a restriction compared to SMTLIB?

Reply:
This is to be determined.  Andres and I decided that we should (at the moment) not support this, since it is significantly more complicated to support and is sometimes nonsensical, since :named within quantifier bodies should never be supported.